Source/WebCore/css/CSSCounterStyleRegistry.cpp

```diff
@@ -114,30 +126,33 @@ RefPtr<CSSCounterStyle> CSSCounterStyleRegistry::decimalCounter()
     return nullptr;
 }
 
-RefPtr<CSSCounterStyle> CSSCounterStyleRegistry::counterStyle(const AtomString& name)
+// A valid map means that the search begins at the author counter style map, otherwise we skip the search to the UA counter styles.
+RefPtr<CSSCounterStyle> CSSCounterStyleRegistry::counterStyle(const AtomString& name, CounterStyleMap* map)
 {
     if (name.isEmpty())
         return decimalCounter();
 
-    auto getCounter = [&](const CounterStyleMap& map, const AtomString& counterName) {
+    auto getCounter = [&](const AtomString& counterName, const CounterStyleMap& map) {
         auto counterIterator = map.find(counterName);
         return counterIterator != map.end() ? counterIterator->value.get() : nullptr;
     };
 
-    auto authorCounter = getCounter(m_authorCounterStyles, name);
-    if (authorCounter)
-        return authorCounter;
-    auto userAgentCounter = getCounter(userAgentCounterStyles(), name);
-    if (userAgentCounter)
-        return userAgentCounter;
-
-    return decimalCounter();
+    // If there is a map, the search starts from the given map.
+    if (map) {
+        auto counter = getCounter(name, *map);
+        if (counter)
+            return counter;
+    }
+    // If there was no map (called for user-agent references resolution), or the counter was not found in the given map, we search at the user-agent map.
+    auto& userAgentMap = userAgentCounterStyles();
```

Comment:
You probably don't need the intermediary variable here, you can use it directly in `getCounter`